src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/ProjectServices/VCProjectSystemServices.cs

```diff
@@ -0,0 +1,407 @@
+        }
+
+        private IEnumerable<LibraryDependency> GetPackageReferencesAsync(
+            MicrosoftBuildEvaluationProject msBuildEvaluationproject,
+            NuGetFramework targetFramework)
+        {
+            var packageReferences = msBuildEvaluationproject.GetItems("PackageReference");
+
+            if (packageReferences != null && packageReferences.Count != 0)
+            {
+                var references = packageReferences.Select(installedPackage =>
+                {
+                    List<string> metadataElements = new List<string>();
+                    List<string> metadataValues = new List<string>();
+
+
+                    foreach (var item in installedPackage.Metadata)
+                    {
+                        if (item.Name.Equals("Version", StringComparison.OrdinalIgnoreCase) == false)
+                        {
+                            metadataElements.Add(item.Name);
+                            metadataValues.Add(item.EvaluatedValue);
+                        }
+                    }
+
+
+                    return new PackageReference(
+                            name: installedPackage.EvaluatedInclude,
+                            version: installedPackage.GetMetadataValue("Version"),
+                            metadataElements: metadataElements.ToArray(),
+                            metadataValues: metadataValues.ToArray(),
+                            targetNuGetFramework: targetFramework);
+
+                })
+                .Where(p => p != null)
+                .Select(ToPackageLibraryDependency);
+
+                return references;
+            }
+
+            return null;
+        }
+
+        //Have PackageReference?
+        public static bool HasPackageReference(IVsProjectAdapter _vsProjectAdapter)
+        {
+            Assumes.Present(_vsProjectAdapter);
+
+            
+
+            return NuGetUIThreadHelper.JoinableTaskFactory.Run(async delegate
+            {
+                await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+                bool bHasPackageReference = false;
+
+                await ProjectHelper.DoWorkInReadLockAsync(
+                    _vsProjectAdapter.Project,
+                    _vsProjectAdapter.VsHierarchy,
+                    buildProject =>
+                    {
+                        var packageReferences = buildProject.GetItems("PackageReference");
+
+                        bHasPackageReference = packageReferences != null && packageReferences.Count != 0;
+                    });
+
+                return bHasPackageReference;
+            });
+        }
+
+        public async Task<IEnumerable<ProjectRestoreReference>> GetProjectReferencesAsync(
+            Common.ILogger _, CancellationToken __)
+        {
+            await _threadingService.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+            //C++ Project not support Reference
+            return new ProjectRestoreReference[] { };
```

Comment:
does this refer to VC project-project references?  What functionality is this omitting?

Reply:
Because native C++ does not support assembly references, so I returned an empty list.